### ISSUE TITLE
The compiler allows unsafe casting, but it should produce an warning if you do one.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#88]
```
* the default annotation is @Mutable  (to make sure existing code would 
compile).
* The compiler allows unsafe casting, but it should produce an warning if you 
do one.


void foo(@ReadOnly Object roObj, @Mutable Object mutObj) {
  @Mutable Date mutDate= (@Mutable Date) roObj; // should produce a warning that we are performing an unsafe cast
  @Mutable Date mutDate= (Date) roObj; // same, because the default is @Mutable
  @Mutable Date mutDate= (Date) mutObj; // shouldn't produce any warnings
}
```

Original issue reported on code.google.com by `Yoav.Zibin@gmail.com` on 12 Nov 2010 at 2:56
